### ISSUE TITLE
DBCluster unable to update values with MasterUserPassword error

### ISSUE THREAD

**Issue by dhumphries-sainsburys** [crossplane-contrib/provider-aws#1807]
### What happened?
Can successfully deploy DBCluster using the below with no problems
```apiVersion: rds.aws.crossplane.io/v1alpha1
kind: DBCluster
metadata:
  name: example-aurora-postgres-cluster
spec:
  forProvider:
    region: eu-west-1
    engine: aurora-postgresql
    engineVersion: '13.6'
    allowMajorVersionUpgrade: true
    backupRetentionPeriod: 14
    copyTagsToSnapshot: true
    masterUsername: masteruser
    manageMasterUserPassword: false
    port: 5432
    autogeneratePassword: true
    autoMinorVersionUpgrade: false
    masterUserPasswordSecretRef:
      name: example-aurora-mysql-cluster
      namespace: crossplane-system
      key: password
    storageEncrypted: true
    databaseName: auroradb
    dbClusterParameterGroupName: example-clusterparametergroup13
    dbSubnetGroupName: lab-data
    skipFinalSnapshot: true
    applyImmediately: true
    finalDBSnapshotIdentifier: test-dan-humphries-lab-aurora-dantest-psql-final
    vpcSecurityGroupIDRefs:
      - name: 'lab-rds-sg'
  providerConfigRef:
    name: aws
  providerRef:
    name: aws
  writeConnectionSecretToRef:
    name: test-dan-humphries-rds-dantest-psql-auroracluster
    namespace: flux-system
```
If i try to update the fields though it returns the below.
```
update failed: cannot update DBCluster in AWS: InvalidParameterCombination: MasterUserPassword must be provided when ManageMasterUserPassword is set to false.\n\tstatus code: 400, request id:
```
The fields i have seen issues with so far are dbClusterParameterGroupName & engineVersion both of which would stop us being able to upgrade clusters. 

The error makes little sense considering we have an autogenerated password anyways and the field it is complaining about doesn't exist in the spec. Also testing this against a cluster where we provide the password as per the example CRs but that gets the same

### How can we reproduce it?
Create the example above and try and change one of the mentioned fields such as putting engineVersion to 13.7

### What environment did it happen in?
Crossplane version: v1.12.1
AWS Provider version: v0.40.0
EKS version: 1.25.11

**Comment by MisterMX** [comment]
This might be solved by #1756?

**Comment by dhumphries-sainsburys** [comment]
Potentially. It seems to be in the same kind of area so might solve this as a side effect but guess we will see when that change gets merged

**Comment by MisterMX** [comment]
@dhumphries-sainsburys can you check whether your problem is solved in `v0.42.0`?

**Comment by dhumphries-sainsburys** [comment]
Can confirm this is fixed in v0.42.0
